### PR TITLE
ENG-17013 - 9.1 -Fix Crash When Companion Stream Dropped in non-stream Catalog Update

### PR DESCRIPTION
**Pull request by ndenggg**

### CODE REVIEW

File: src/ee/execution/VoltDBEngine.cpp [hunk 1]
```diff
@@ -1217,7 +1217,7 @@ VoltDBEngine::processCatalogAdditions(int64_t timestamp, bool updateReplicated,
             }
             if (streamedTable) {
                 //Dont update and roll generation if this is just a non stream table update.
-                if (isStreamUpdate) {
+                if (isStreamUpdate || persistentTable->getStreamedTable()) {
                     const std::string& name = streamedTable->name();
```
Comment (zeemanhe) [inline src/ee/execution/VoltDBEngine.cpp]: Can you add comments here to explain the issue?

Comment (ndenggg) [reply src/ee/execution/VoltDBEngine.cpp]: Updated, thx

File: src/ee/execution/VoltDBEngine.cpp [hunk 2]
```diff
@@ -1218,7 +1218,6 @@ VoltDBEngine::processCatalogAdditions(int64_t timestamp, bool updateReplicated,
             if (streamedTable) {
                 // Updated in ENG-17013: when UAC handles non-stream change, EE will attach the stream
                 // when rebuilding migrate table so that it won't lose its stream wrapper
-                // So far this caused many test failure
                 if (isStreamUpdate || (persistentTable && persistentTable->getStreamedTable())) {
```
Comment (JadenW) [inline src/ee/execution/VoltDBEngine.cpp]: Feels like this is a special case for migrate table's companion stream. So we can move this logic to Line 1211 which specifically state the situation).

Comment (ndenggg) [reply src/ee/execution/VoltDBEngine.cpp]: Will update in master's pr, thx

**Pull request merged**